### PR TITLE
bpf: Avoid unnecessary debug output on policy map open

### PR DESCRIPTION
**Pull request by tgraf**

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

File: pkg/bpf/map.go [hunk 1]
```diff
@@ -512,3 +514,23 @@ func ConvertKeyValue(bKey []byte, bValue []byte, key interface{}, value interfac
 
 	return nil
 }
+
+// MetadataDiff compares the metadata of the BPF maps and returns false if the
+// metadata does not match
+func (m *Map) MetadataDiff(other *Map) bool {
+	if m == nil {
```
Comment (aanm) [inline pkg/bpf/map.go]: check if `other` is also nil

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by tgraf** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: pkg/bpf/map.go [hunk 2]
```diff
@@ -512,3 +514,23 @@ func ConvertKeyValue(bKey []byte, bValue []byte, key interface{}, value interfac
 
 	return nil
 }
+
+// MetadataDiff compares the metadata of the BPF maps and returns false if the
+// metadata does not match
+func (m *Map) MetadataDiff(other *Map) bool {
+	if m == nil || other == nil {
```
Comment (raybejjani) [inline pkg/bpf/map.go:521]: Oh, would both being nil be a `true`?

Comment (tgraf) [reply pkg/bpf/map.go:521]: Thought about it, doesn't really make sense.

**Review by borkmann — APPROVED** [review]

**Pull request merged**